From 294ad989836dc9e3f235ed0abcc71c1b6cdb881a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Wed, 19 Nov 2014 11:02:06 -0500
Subject: [ethernet] i40e: disable FCoE

Message-id: <546C78AE.6080407@redhat.com>
Patchwork-id: 100316
O-Subject: [RHEL7.1 PATCH] i40e: disable FCoE
Bugzilla: 1165175
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>

Bugzilla:
https://bugzilla.redhat.com/show_bug.cgi?id=1165175

Description:
It was decided to disable i40e FCoE for now. Adding a return 0 to the
beginning of function i40e_init_pf_fcoe() to make sure it does not
get enabled even if the hardware supports it.

Signed-off-by: Stefan Assmann <sassmann@redhat.com>

Upstream Status:
Not upstream, temporary workaround for RHEL7.1.

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8261159

Test:
Tested by myself. No i40e FCoE netdev is created with the patch applied.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_fcoe.c b/drivers/net/ethernet/intel/i40e/i40e_fcoe.c
index 5d01db1..dc37b77 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_fcoe.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_fcoe.c
@@ -287,6 +287,10 @@ int i40e_init_pf_fcoe(struct i40e_pf *pf)
  struct i40e_hw *hw = &pf->hw;
  u32 val;
 
+ /* RHEL7.1 has FCoE disabled */
+ dev_info(&pf->pdev->dev, "FCoE capability is disabled\n");
+ return 0;
+
  pf->flags &= ~I40E_FLAG_FCOE_ENABLED;
  pf->num_fcoe_qps = 0;
  pf->fcoe_hmc_cntx_num = 0;
-- 
1.7.1